src/middlewared/middlewared/plugins/disk.py

```diff
@@ -112,6 +115,23 @@ def _compress_enclosure(self, disk):
             disk['enclosure_slot'] = None
         del disk['enclosure']
 
+    @filterable
+    async def query_passwords(self, filters=None, options=None):
+        """
+        Query disks SED passwords with `query-filters` and `query-options`.
+        """
+        disks = await self.middleware.call(
+            'datastore.query', self._config.datastore, filters, {**options, 'prefix': self._config.datastore_prefix}
+        )
+        disks_keys = await self.middleware.call('kmip.retrieve_sed_disks_keys')
+        for disk in ([disks] if not isinstance(disks, list) else disks):
```

Comment:
Why would disks not be a list?

Reply:
If `get=True` then it would not be a list